synapse/static/client/login/js/login.js

```diff
@@ -50,8 +43,9 @@ var setFeedbackString = function(text) {
 };
 
 var show_login = function(inhibit_redirect) {
-    var this_page = window.location.origin + window.location.pathname;
-    $("#sso_redirect_url").val(this_page);
+    // Set the redirect to come back to this page, a login token will get added
+    // and handled after the redirect.
+    $("#sso_redirect_url").val(window.location.href);
```

Comment:
This change means that we use the whole URL as the redirect, which is necessary to pass the query parameters through SSO.

Reply:
passing the query params through SSO feels like the sort of thing we shouldn't be doing, for fear of CSRF. can we stash them in a cookie or something instead?

Reply:
Hmm...the concern being the request will be modified before it gets back to us? I think that would be "bad", but I don't see a security issue with it. We can probably save it in a cookie though. I'll take a look!